fe/fe-core/src/main/java/com/starrocks/scheduler/history/TaskRunHistoryTable.java

```diff
@@ -210,8 +210,17 @@ public List<TaskRunStatus> lookup(TGetTasksParams params) {
             predicates.add(" task_state = " + Strings.quote(params.getState()));
         }
         sql += Joiner.on(" AND ").join(predicates);
-        if (params.isSetPagination() && params.getPagination().getLimit() > 0) {
-            sql += " LIMIT " + params.getPagination().getLimit();
+
+        if (params.isSetPagination()) {
+            if (params.getPagination().getLimit() > 0 || params.getPagination().getOffset() > 0) {
+                sql += " ORDER BY `task_id`, `task_run_id`, `create_time`";
+            }
+            if (params.getPagination().getLimit() > 0) {
+                sql += " LIMIT " + params.getPagination().getLimit();
```

Comment:
Can `Pagination` cause result duplication or unstable if history table is changing?

Reply:
Another question, can we always add a `limit` for the history table to avoid fetching too many results?

Reply:
1. Will it cause duplication? Technically, no. The `task_runs` consists of three components: running, in-memory, and in-table. Therefore, it can be enumerated using `offset + limit`.  
2. Is it stable? Not entirely, but it’s not a critical issue.  
3. How? If we want to retrieve all records, we can query something like: select count(*) from `information_schema.task_runs`.  Then you cannot add an implicit limit for the query